### PR TITLE
Opium 3.8

### PR DESCRIPTION
**Pull request by davydden**
Opium is a software to generate pseudopotentials for ab initio calculations in condensed matter physics, computational chemistry and material science.

### CODE REVIEW

**Comment by dpo** [comment]
Any idea why the test fails on Yosemite?

**Comment by davydden** [comment]
does it run completely for you?

For me I have differences in `fe.cpi`, and `pt.cpi`. I wrote to the mailing list to if it is indeed some issue or `blessed` files are not up-to-date

**Comment by davydden** [comment]
Note that also removed compiling `lapack` and substituted it with `veclibfort`, but this shouldn't be an issue anyway.

**Comment by dpo** [comment]
It builds fine for me under 10.9 and 10.10, and the tests run, but I also get
````
==> +++  File "fe.cpi" differs from file "results/fe.cpi"
+++  File "pt.cpi" differs from file "results/pt.cpi"
````
on both platforms.

**Comment by davydden** [comment]
well, i asked developers, so far no answer: https://sourceforge.net/p/opium/mailman/message/33287478/

**Comment by davydden** [comment]
just contacted the project coordinator E.J. Walter, hope there will be a resolution to this issue.

**Comment by davydden** [comment]
Here is the comment from the project coordinator E.J.Walter:
>The differences you see for fe.cpi and pt.cpi are most likely due to the use of gfortran on the mac.  My tests use f2c + fort on the mac since the gfortran compiler (on both mac and linux) is still a little unstable and causes some (usually small) numerical differences to creep up.  If you run configure and gfortran is found as the only fortran compiler a warning stating this is issued.
>
>I am a little surprised by the size of the of the error for fe.cpi.  
>
>I would be very happy to see the opium package to be part of homebrew-science.  If I get some time, I will look into the numerical issues surrounding gfortran on mac and linux.

so i guess i will modify Caveats accordingly and it should be good to go.

**Comment by dpo** [comment]
Not sure what he means by "fort" or by "gfortran ... is still a little unstable".

**Comment by davydden** [comment]
i suppose what he means is that `gfortran`, being a free compiler, is not as reliable to produce the correct machine code as, say, `ifort` (Intel Fortran Compiler), which you have to buy. It is also related to which implementation of BLAS/Lapack is used.

File: opium.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
+class Opium < Formula
+  homepage "http://opium.sourceforge.net"
+  url "https://downloads.sourceforge.net/project/opium/opium/opium-v3.8/opium-v3.8-src.tgz"
+  sha1 "9649e79e2393fe35e7575ab31dffd2b1f0437e71"
```
Comment (dpo) [inline opium.rb]: Would you mind updating this to a `sha256`?

File: opium.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  patch :DATA
+
+  def install
+    system "./configure", "LDFLAGS=-lveclibfort"
+    cd "src"
+    system "make", "all-subdirs"
+    system "make", "opium"
+    cd ".."
+    bin.install "opium"
+    (share/"opium-testdir").install Dir["testdir/*"]
```
Comment (dpo) [inline opium.rb]: Does `opium` not have a `make install`?

Comment (davydden) [inline opium.rb]: funny enough - not.

**Comment by dpo** [comment]
@BrewTestBot test this please

**Comment by davydden** [comment]
I am working on a Formula for `ape` pseudo-potential project http://www.tddft.org/programs/APE/ (it's also on MacPorts https://trac.macports.org/browser/trunk/dports/science/ape/Portfile). It does pretty much the same thing (at least to my understanding), but looks like a more active project. The current version does not compile against `libxc` 2.2.0. I notified developers and the promised to have a look at it. `homebrew-science` could have both and remove either if it is not used much.

**Comment by dpo** [comment]
Should this still be pulled, or is it superseded by `ape`?

**Comment by davydden** [comment]
I think this PR can be closed without pulling. I will keep the branch, just in case...

**Comment by dpo** [comment]
Ok thanks. Feel free to reopen if need be.

**Pull request closed without merging**